nullaway/src/main/java/com/uber/nullaway/handlers/StreamNullabilityPropagator.java

```diff
@@ -204,14 +217,62 @@ public void onMatchMethodInvocation(
             ClassTree annonClassBody = ((NewClassTree) argTree).getClassBody();
             // Ensure that this `new B() ...` has a custom class body, otherwise, we skip for now.
             if (annonClassBody != null) {
-              MaplikeMethodRecord methodRecord = streamType.getMaplikeMethodRecord(methodSymbol);
-              handleMapAnonClass(methodRecord, tree, annonClassBody);
+              MapLikeMethodRecord methodRecord = streamType.getMaplikeMethodRecord(methodSymbol);
+              handleMapOrCollectAnonClassBody(
+                  methodRecord,
+                  annonClassBody,
+                  t -> observableCallToInnerMethodOrLambda.put(tree, t));
             }
           } else if (argTree instanceof LambdaExpressionTree) {
             observableCallToInnerMethodOrLambda.put(tree, argTree);
           } else if (argTree instanceof MemberReferenceTree) {
             observableCallToInnerMethodOrLambda.put(tree, argTree);
           }
+        } else {
+          CollectLikeMethodRecord collectlikeMethodRecord =
+              streamType.getCollectlikeMethodRecord(methodSymbol);
+          if (collectlikeMethodRecord != null && methodSymbol.getParameters().length() == 1) {
+            handleCollectCall(tree, collectlikeMethodRecord);
+          }
```

Comment:
Not for this PR, but wonder if as a follow up you want to extract the other two cases of this if into their own methods too, with `onMatchMethodInvocation(...)` as just a dispatcher.